CMakeLists.txt

```diff
@@ -31,13 +31,15 @@ add_executable(pong ${SOURCES})
 add_subdirectory(glfw-3.3.8)
 add_subdirectory(glew-2.1.0)
 add_subdirectory(glm-0.9.9.8)
+add_subdirectory(json-3.11.2/json-3.11.2)
```

Comment:
Curious what is the best way to organize tps libraries for this project @cjappl? I know we used zip files in some way before? Looking to include these in a nice way, that doesn't make this project show as being mainly an HTML repo haha

Reply:
So this is a great time to reorganize your tps. I highly recommend you zip all of these dependencies into tar.gz files to save space (and ensure you don't have some crazy 2k line PR). I would recommend you add a `tps` top level folder with it's own CMakeLists.txt

From there, the pattern is like so in your tps/CMakeLists.txt:

```
FetchContent_Declare(doctest
    URL ${CMAKE_CURRENT_SOURCE_DIR}/doctest/doctest-2.4.11.tar.gz
)
FetchContent_MakeAvailable(doctest)

```

What this does is automagically unzip/untar the thing, then `FetchContent_MakeAvailable` goes through and calls `add_subdirectory` like you are doing here

If you ever want to play around with a dependency without adding it to your repo, or you don't want to check something in, FetchContent works on git repos that are built with CMake:

```
# Fetch zlib from its Git repository
FetchContent_Declare(
    zlib
    GIT_REPOSITORY https://github.com/madler/zlib.git
    GIT_TAG v1.2.11  # You can specify a specific tag or commit hash
)
```

Reply:
Yup that is much nicer, thanks! Ill follow up with the rest of the tps libraries